### ISSUE TITLE
Frontend caching all last queried balances and display on restart

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#1706]
## Abstract

This is a secondary issue derived from the talk with Jez on  https://github.com/rotki/rotki/issues/1705.

The goal of the issue is to see if it's possible for the frontend's state to be cached once all balances have been queried and to display that state immediately upon restart with an indicator that new balances are loading.

## Motivation

The goal is to have a much better UX by not having the user wait so long until they see something when they start the application (some times minutes if they got hundreds of accounts).

## Task

Do it and probably @kelsos spec it out and tell us if it's possible

**Comment by kelsos** [comment]
An idea is to use to put snapshots of the frontend balance state (vuex) in a key in the database (maybe frontend settings or a new key) and restore that on login. The pieces needed for this to work should already be already there.

**Comment by kelsos** [comment]
So as per our discussion on discord we should see if there is a way to get the last fetched responses from the backend when the user logs in and before doing the big call to fetch all balances.

The idea would be for the backend to provide an option to get the latest balances snapshot. Upon login, the frontend can call this set up the state and then do the balance saving call to update to the latest.

**Comment by kelsos** [comment]
I was thinking that if we remove the reliance on the big balance query this should speed up things a lot for the dashboard to. 
So maybe this won't be necessary anymore.

**Comment by LefterisJP** [comment]
This is quite possible. So you are referring to the iterative query, right? So when that goes in this won't be needed. So let's close this: https://github.com/rotki/rotki/issues/2124

**Comment by kelsos** [comment]
Yes, the problem here is that if you have many accounts and the backend takes a long period of time to respond you see nothing.
If we move to iterative and then the frontend gets data faster then this won't be a problem.

**Comment by kelsos** [comment]
As we discussed I am re-opening this.

The approach for this would be for the backend to keep a cached snapshot of the last known balances. 

The frontend can request this snapshot to display before it starts the refresh/slow loading of the fresh balances.

**Comment by kelsos** [comment]
@LefterisJP So this is the issue I think. About storing balances in the database.

**Comment by kelsos** [comment]
OK So like we discussed:

- This new table can take the role of an upgraded snapshot table that also contains a location
- The old snapshot table will remain read-only since it has valuable data but lacks location information we cannot infer.
- An extra table we will add is the one that will create balances based on the historic chain state.

For graphs and historical data, we can use a combination of these tables in the future

**Comment by Hugo0** [comment]
user here: this would improve UX a lot

**Comment by yabirgb** [comment]
Blockchain balances will be cached in 1.43 https://github.com/rotki/rotki/pull/11979
